core/tests/puppeteer-acceptance-tests/utilities/user/curriculum-admin.ts

```diff
@@ -807,6 +703,274 @@ export class CurriculumAdmin extends BaseUser {
+      const skillNameElement = await skill.$(skillNameSelector);
+      if (!skillNameElement) {
+        return false;
+      }
+      const name = await (
+        await skillNameElement.getProperty('textContent')
+      ).jsonValue();
+      return name === skillName;
+    });
+
+    if (!skillToDelete) {
+      throw new Error(`Skill ${skillName} not found`);
+    }
+
+    await this.clickElement(skillToDelete, skillListItemOptions);
+    await this.page.waitForFunction(
+      (skill: Element) => !skill.isConnected,
+      {},
+      skillListItemOptions
+    );
+    await this.clickElement(skillToDelete, deleteSkillButton);
+    await this.page.waitForFunction(
+      (button: Element) => !button.isConnected,
+      {},
+      deleteSkillButton
+    );
+    await this.clickElement(this.page, confirmSkillDeletionButton);
+    await this.page.waitForFunction(
+      (button: Element) => !button.isConnected,
+      {},
+      confirmSkillDeletionButton
+    );
+  }
+
+  /**
+   * Function to check if a skill is not present in the Topics and Skills Dashboard.
+   * @param {string} skillName - The name of the skill to check.
+   */
+  async expectSkillNotInTopicsAndSkillsDashboard(
+    skillName: string
+  ): Promise<void> {
+    await this.goto(topicAndSkillsDashboardUrl);
+    const isTextPresent = await this.isTextPresentOnPage(
+      ' No topics or skills have been created yet. '
+    );
+    if (isTextPresent) {
+      showMessage(`The skill "${skillName}" is not present on the Topics and Skills
+      Dashboard as expected.`);
+      return;
+    }
+    await this.clickOn(skillsTab);
+    const isSkillPresent = await this.isTextPresentOnPage(skillName);
+    if (isSkillPresent) {
+      throw new Error(
+        `Skill "${skillName}" was found.
+          It was expected to be absent from Topics and Skills Dashboard.`
+      );
+    }
+    showMessage(
+      `The skill "${skillName}" is not present on the Topics and Skills
+      Dashboard as expected.`
+    );
+  }
+
+  /**
+   * Function to delete all questions in a skill.
+   * @param {string} skillName - The name of the skill to delete questions from.
+   */
+  async removeAllQuestionsFromTheSkill(skillName: string): Promise<void> {
+    await this.openSkillEditor(skillName);
+
+    const isMobileWidth = this.isViewportAtMobileWidth();
+    const skillQuestionTab = isMobileWidth
+      ? mobileSkillQuestionTab
+      : desktopSkillQuestionTab;
+
+    if (isMobileWidth) {
+      await this.clickOn(mobileOptionsSelector);
+      await this.page.screenshot({path: 'debug1.png'});
+      await this.page.waitForSelector(
+        '.e2e-test-mobile-skill-nav-dropdown-icon',
+        {visible: true}
+      );
+      await this.clickOn('.e2e-test-mobile-skill-nav-dropdown-icon');
+    }
+    await this.clickAndWaitForNavigation(skillQuestionTab);
+
+    let isTextPresent = await this.isTextPresentOnPage(
+      'There are no questions in this skill.'
+    );
+
+    while (!isTextPresent) {
```

Comment:
If the text is never present, then this will be an infinite loop. Is it intended to be a while loop? I think it should be if condition instead of while loop.

Reply:
Actually, it is intended, this is because this function intendeds to remove all the questions (that's why can't use if..else) from a skill and we don't know number of questions present in the skill (To keep the function general).
Additionally, if the text is never present, it only means that the page has more questions to be removed and if anything goes wrong with the removal than the loop will terminate and throw appropriate error. So, It won't end up into a infinite loop.

Reply:
I see, thanks for explaining to me!